rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,429 @@
+
+## Changes the existing comments inside the code base
+
+This could be automated.
+
+Also, the migration could be performed piecemal, starting perhaps with `nixpkgs.lib`.
+
+See future work.
+
+## Tooling to produce the nixpkgs manual
+
+Change or write a new tool used to produce Nixpkgs library function documentation. Because this is a breaking change.
+
+See future work.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## All considered outer formats
+
+| Property / Approach | `##` | `/** */` | `Javadoc` | `/*\|` or `/*^`  | All comments are doc-comments |
+|---|---|---|---|---|---|
+| Inspired by | Rust | Current nixpkgs.lib | C++/Java/Javascript | Haskell Haddock | Current nixpkgs.lib |
+| Changes the existing code by | Much | Less | Even More | Less | ? |
+| Needs Termination | No | Yes | Yes | Yes | ? |
+| Indentation | Clear | like Nix's multiline strings, thus **Intuitive** | Clear | ? | ? |
+| Needs vertical space  | No | Yes | Yes | Yes | ? |
+| Visual distinction from comments | High | Low | Medium | Medium | No distinction at all |
+| Needs Autocompletion (Language Server) to continue the next line. | Yes | No | Yes | No | ? |
+| Punctuation Variations / Amount of different special characters | 1 (Less) | 2 (Medium) | 2 (Medium) | 3 (More) | None |
+| Markdown compatibility (also depends on indentation clarity) | **Visual conflicts with headings `# Title`** | Good | Medium | Good | Only with multiline comments |
+| breaks when interrupted with newlines | Yes | No | ? | No | |
+| Simplicity (Brainload) | Medium | Simple | Complex | More Complex | |
+
+### Refactoring note
+
+**Observing**: From a refactoring perspective, it might also be interesting to see how many conflicts the different formats would cause.
+
+nixpkgs comments:
+
+- `##` ~4k usages
+- `#` ~20k usages
+- `/*` ~6k usages
+- `/**` 160 usages 
+
+Choosing `/**` or subsets would cause minor conflicts within current nixpkgs. While this is NOT the main reason for the final decision, it MUST be considered.
+
+## Just free text as a content format
+
+While this allows the most freedom, it is usually considered the best option, not creating any restrictions.
+
+- [RFC72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md) defines commonMark as the official documentation format.
+This is why we decided to follow this convention. Writing plain text is still possible.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Migration path for nixpkgs comments.
+- How to document the `arguments`. Should there be some markdown equivalent to `@param` i.e. `# Params`?
+    - Possible answer: Future RFC might need to clarify that.
+
+# Future work
+[Future]: #future-work
+
+## Migrate existing nixpkgs comments
+
+Reformatting existing doc-comments in nixpkgs.
+
+## nixpkgs Manual
+
+The current nixpkgs manual needs to be adopted to this change.
+
+## Native support in Nix
+
+- `NixOS/nix` should implement native support for doc-comments.
```

Comment:
Can we make a clearer argument of why Nix should do this? Otherwise, I'd make a weaker claim:

```suggestion
There have been efforts to add support for parsing doc comments to the Nix evaluator. These could be revived and adapted to the format proposed by this RFC.
```